fedimint-derive/src/lib.rs

```diff
@@ -121,7 +121,16 @@ pub fn derive_encodable(input: TokenStream) -> TokenStream {
             }
         }
         syn::Data::Enum(DataEnum { variants, .. }) => {
-            let match_arms = variants.iter().enumerate().map(|(variant_idx, variant)| {
+            if variants.is_empty() {
+                quote! {
+                    impl Encodable for #ident {
+                        fn consensus_encode<W: std::io::Write>(&self, _writer: &mut W) -> std::result::Result<usize, std::io::Error> {
+                            unreachable!("Enum without variants can't exist")
```

Comment:
```suggestion
                            match self { }
```

Reply:
LGTM, but this one worth fixing before landing.

Reply:
Done. Most interestingly I had to dereference the enum or [I would get this error](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021):

![Screenshot 2023-01-19 at 14-05-56 Rust Playground](https://user-images.githubusercontent.com/84478054/213548719-e9629dbb-528e-4fa4-839e-889cbc6b924b.png)